### PR TITLE
[BX-1314][BX-1315] Support Additional Simplehash Networks + Configure Testnet NFT Request

### PR DESCRIPTION
**Pull request by derHowie**
## What changed (plus any additional context for devs)
Added support for all available networks returned by Simplehash that also overlapped with Viem's supported network list. Partitioned mainnet and testnet nft network lists and now return testnet NFTs in dev mode. 

## Screen recordings / screenshots
PoW: https://recordit.co/jowwjwvIp2

## What to test
Ensure NFTs on networks that are currently unsupported, but added to the Simplehash request in this PR appear. Also ensure that testnet nfts only appear while dev mode is activated.

### CODE REVIEW

File: src/core/resources/nfts/nfts.ts [hunk 1]
```diff
@@ -56,14 +55,9 @@ async function nftsQueryFunction({
...
-  const chains = (
-    !testnetMode ? SUPPORTED_MAINNET_CHAINS : SUPPORTED_TESTNET_CHAINS
-  )
-    .map(({ name }) => name as ChainName)
-    .filter((chainName) =>
-      simpleHashSupportedChains.includes(chainName.toLowerCase()),
-    );
+  const chains = !testnetMode
+    ? (getSimpleHashSupportedChainNames() as ChainName[])
+    : (getSimpleHashSupportedTestnetChainNames() as ChainName[]);
```
Comment (estebanmino) [inline src/core/resources/nfts/nfts.ts:60]: this means that we're not filtering nfts by enabled chains from settings anymore?

Comment (estebanmino) [reply src/core/resources/nfts/nfts.ts:60]: oh wait, do we even filter by enabled chain?

**Comment by derHowie** [comment]
@estebanmino https://rainbowhaus.slack.com/archives/C044WE28BTK/p1707492095101959?thread_ts=1707450430.132609&cid=C044WE28BTK

File: src/entries/popup/pages/home/NFTs/NFTDetails.tsx [hunk 2]
```diff
@@ -626,8 +627,9 @@ const NFTAccordionAboutSection = ({
   nft?: UniqueAsset | null;
   showFloorPriceExplainerSheet: () => void;
 }) => {
-  const network =
-    nft?.network === 'mainnet' ? 'Ethereum' : capitalize(nft?.network);
+  const networkDisplay = nft?.network
+    ? ChainNameDisplay[chainIdFromChainName(nft?.network)]
+    : '';
```
Comment (greg-schrammel) [inline src/entries/popup/pages/home/NFTs/NFTDetails.tsx:632]: is this `''` just used as a "falsy" value?

**Review by estebanmino — APPROVED** [review]

File: src/entries/popup/pages/home/NFTs/NFTDetails.tsx (hunk 2 above)
Comment (derHowie) [reply src/entries/popup/pages/home/NFTs/NFTDetails.tsx:632]: Yes, just not showing that row if nft is undefined. Just being safe, that shouldn't be the case unless something broke.

**Review by greg-schrammel — APPROVED** [review]

**Pull request merged**